src/build/index.js

```diff
@@ -33,13 +33,25 @@ module.exports = async (argv) => {
     env: {
       NODE_ENV: process.env.NODE_ENV || 'production',
       AEGIR_BUILD_ANALYZE: argv.bundlesize,
-      AEGIR_NODE: argv.node
+      AEGIR_NODE: argv.node,
+      AEGIR_TS: argv.ts
     },
     localDir: path.join(__dirname, '../..'),
     preferLocal: true,
     stdio: 'inherit'
   })
 
+  if (argv.ts) {
+    await execa('tsc', [
+      '--outDir', './dist/src',
+      '--declaration'
```

Comment:
you should ass `--sourceMap` as it will provide info where error occured in typescript source

Reply:
ah right, whats the use case for this? local aegir bundling (build and test) already has sourcemaps, so this would be just for the final users.
- in node they would need to explicitly run with --enable-source-maps true
- in the browser we don't really run any error tracing code 
So the only thing that i see is a user making an issue with a stack trace, which would be easy to trace since we publish dist/src mostly untouched (besides cjs<>esm crap), plus most users dont include sourcemaps from deps.

An example would be me debugging libp2p-noise from the js-ipfs, i wouldn’t get the sourcemaps because i wouldn’t run with --enable-source-maps true in node and in the browser aegir doesnt load deps sourcemaps.

Reply:
TBH not sure if its gonna work in normal console but most IDEs detect those in node modules and display links to typsecript source.

Reply:
Really? my experience with TS in vscode is that you always get links to .d.ts and not source files, makes me mad. :)

Reply:
> Really? my experience with TS in vscode is that you always get links to .d.ts and not source files, makes me mad. :)

:rofl: It's far from perfect! 

I just keep it and ship it because someone might know to handle them. Either way, it can be specified per project in tsconfig so I'm fine either way :smile: